migrations/V122__Update_PlanBasics_ModelType_MultiSelect_And_Other_Text.sql

```diff
@@ -0,0 +1,40 @@
+ALTER TYPE MODEL_TYPE RENAME TO MODEL_TYPE_OLD;
+
+-- Add the 'OTHER' and 'MANDATORY_REGIONAL_OR_STATE' values to the old type so we can migrate 'TBD' and 'MANDATORY'
+-- selections later in this migration
+ALTER TYPE MODEL_TYPE_OLD ADD VALUE 'OTHER';
+ALTER TYPE MODEL_TYPE_OLD ADD VALUE 'MANDATORY_REGIONAL_OR_STATE';
+
+
+CREATE TYPE MODEL_TYPE AS ENUM (
+  'VOLUNTARY',
+  'MANDATORY_REGIONAL_OR_STATE',
+  'MANDATORY_NATIONAL',
+  'OTHER'
+);
+
+COMMIT;

```

Comment:
@OddTomBrooks, was `POSTGres` complaining if you didn't add a commit here? There might be a way around using it. 

I'm a bit uneasy with the commit in the middle. Flyway calls each of these scripts in their own transaction, and rolls them back if there are any error. I don't know what happens if you have already committed partially into the script (does those changes stay?).

I think it would be better to remove the commit.

Reply:
Yes exactly. There was an error trying to use the enums before committing, otherwise it appears the choice is to separate this into two scripts. I'll take another look at this, I remember running into a similar issue before some time ago and I think the resolution was just to make two flyway migrations.

Reply:
👍 , I have also gotten around this before by casting the column to TEXT, doing the update, and then casting to the new type. That should likely resolve the issue, and make sure the whole migration is reverted instead of separating into two migrations.

Reply:
This makes sense. Do you remember the migration? If not, it should be fine, I just want to maintain consistency.

Reply:
Sure! It's in main , V122__Update_Funding_Source_Type.sql

I set the column type to TEXT, then when I'm doing the update, the new values are just text. Then after the text is updated, I cast it to the new type